reducer/TestSimpleReducer.java

```diff
@@ -0,0 +1,141 @@
+package reducer;
+
+import java.util.*;
+import sqlancer.ReducerContext;
+
+public class TestSimpleReducer {
+
+    public static void main(String[] args) throws Exception {
+        System.out.println("=== SimpleReducer Test Suite ===");
+
+        runTest("Exception Reduction", TestSimpleReducer::testException);
+        runTest("NoREC Oracle Reduction", TestSimpleReducer::testNoRECOracle);
+        runTest("TLP WHERE Oracle Reduction", TestSimpleReducer::testTLPWhereOracle);
+
+        System.out.println("\n=== All tests completed successfully ===");
+    }
+
+    private static void runTest(String testName, Runnable test) {
+        System.out.println("\n--- " + testName + " ---");
+        test.run();
+    }
+
+    private static void testException() {
+        try {
+            ReducerContext context = createExceptionContext();
+            executeTest(context);
+        } catch (Exception e) {
+            System.err.println("Exception test failed: " + e.getMessage());
```

Comment:
It's unusual to print information about success and failure in unit tests. It would be better to let the JUnit framework handle this.